hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -148,17 +166,20 @@ private void initializePipelineState() throws IOException {
   }
 
   @Override
-  public synchronized Pipeline createPipeline(
-      ReplicationType type, ReplicationFactor factor) throws IOException {
+  public synchronized Pipeline createPipeline(ReplicationType type,
+      ReplicationFactor factor) throws IOException {
     lock.writeLock().lock();
     try {
       Pipeline pipeline = pipelineFactory.create(type, factor);
       pipelineStore.put(pipeline.getId().getProtobuf().toByteArray(),
           pipeline.getProtobufMessage().toByteArray());
       stateManager.addPipeline(pipeline);
       nodeManager.addPipeline(pipeline);
-      metrics.incNumPipelineCreated();
-      metrics.createPerPipelineMetrics(pipeline);
+      metrics.incNumPipelineAllocated();
+      if (pipeline.isOpen()) {
+        metrics.incNumPipelineCreated();
+        metrics.createPerPipelineMetrics(pipeline);
+      }
```

Comment:
`      metrics.incNumPipelineCreated();
        metrics.createPerPipelineMetrics(pipeline);
`
I guess we should move these inside openPipeline function.

Reply:
This statement is for ONE factor RATIS pipeline which is in OPEN after created.  It doesn't go through openPipeline.

Reply:
Sorry! I did not get this point. createPipeline(ReplicationType type, ReplicationFactor factor) would create ONE factor pipeline also in ALLOCATED state according to the patch. Am I missing something?

Reply:
Sorry,  the if (pipeline.isOpen) logic is for standalone pipeline, not the ONE factor RATIS pipeline. 
All RATIS pipeline follow the same rule.